syntax = "proto3";

package aloha.v1;

// Service definition for Ethereum address operations
service AlohaEthereumService {
  // Endpoint to checksum an Ethereum address
  rpc ChecksumAddress(ChecksumAddressRequest) returns (ChecksumAddressResponse) {
    option idempotency_level = NO_SIDE_EFFECTS;
  }
  
  // Endpoint to return a null address
  rpc GetNullAddress(NullAddressRequest) returns (NullAddressResponse) {
    option idempotency_level = NO_SIDE_EFFECTS;
  }
}

// Request message containing the Ethereum address to be checksummed
message ChecksumAddressRequest {
  string address = 1;
}

// Response message containing the original address and checksummed address
message ChecksumAddressResponse {
  string address = 1;
  string checksummed = 2;
}

// Empty request for getting a null address
message NullAddressRequest {}

// Response containing a null Ethereum address
message NullAddressResponse {
  string address = 1;
  string checksummed = 2;
}